File: block_sites/ruby/block.rb
#!/usr/bin/env ruby

hosts = '/etc/hosts'

lines = []

blocked_sites_section = false

File.open(hosts, 'r') do |f|
  while line = f.gets
    blocked_sites_section = true if line =~ /blocked\ssites/
    blocked_sites_section = false if line =~ /blocked\ssites\send/

    if blocked_sites_section && line =~  /^\s*#+\s*127\.0\.0\.1/
      line.gsub!(/^\s*#+\s*/, '')
    end

    lines << line
  end
end

File.open(hosts, 'w') do |f|
  f.puts lines
end

File: block_sites/elixir/block.exs
#! /usr/bin/env elixir

filename = "/etc/hosts"

File.stream!(filename)
|> Stream.map(&String.trim_leading/1)
|> Stream.transform(false, fn line, blocked_sites_section ->
  if blocked_sites_section do
    line =
      if String.match?(line, ~r/^#+\s*127\.0\.0\.1/) do
        String.replace(line, ~r/^#+\s*/, "")
      else
        line
      end

    blocked_sites_section = !String.match?(line, ~r/blocked sites end/)

    {[line], blocked_sites_section}
  else
    blocked_sites_section = String.match?(line, ~r/blocked sites start/)

    {[line], blocked_sites_section}
  end
end)
|> Enum.to_list()
|> (fn list ->
      file = File.open!(filename, [:write])
      IO.write(file, list)
    end).()
